core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -996,7 +999,8 @@ private ExprValue exprQuarter(ExprValue date) {
    * @return ExprValue.
    */
   private ExprValue exprSecond(ExprValue time) {
-    return new ExprIntegerValue(time.timeValue().getSecond());
+    return new ExprIntegerValue(
+        (SECONDS.between(LocalTime.MIN, time.timeValue()) % 60));
```

Comment:
Does `time.timeValue()` work when arg is `STRING`?

Reply:
Yes. Both `SELECT second_of_minute('12:23:34');` and `SELECT second_of_minute("12:23:34");` work